include/vara/Feature/FeatureModelParser.h

```diff
@@ -64,6 +64,89 @@ class FeatureModelXmlParser : public FeatureModelParser {
   static std::unique_ptr<xmlDtd, void (*)(xmlDtdPtr)> createDtd();
 };
 
+//===----------------------------------------------------------------------===//
+//                         FeatureModelSxfmParser Class
+//===----------------------------------------------------------------------===//
+
+/// \brief Parsers for feature models in SXFM.
+/// The SXFM (simple XML feature model) format is a feature model format embeded
+/// in an XML structure.
+class FeatureModelSxfmParser : public FeatureModelParser {
+public:
```

Comment:
```suggestion
public:
  using UniqueXmlDoc = std::unique_ptr<xmlDoc, void (*)(xmlDocPtr)>;
  using UniqueXmlDtd = std::unique_ptr<xmlDtd, void (*)(xmlDtdPtr)>;

```
Hmm, I think we could add some short cuts for the xml unique_ptrs :)

Reply:
This should be fixed now. Could you check it, please?

Reply:
Maybe it is also a good idea to shift this into the superclass FeatureModelParser itself since the XML implementation could use that too.

Reply:
If it's in general the same that sounds like a good idea.

Reply:
I think we should either introduce a subclass of the FeatureModelParser that puts all commonalities of XML (like) Parser together or introduce these short cut in both (XML and SXFM Parser) as these types seem to be very specific to Parser for XML (-like) FIles, e.g. if we add a feature model parser for a CNF/SMT format, we would not need these short cuts, right?